framework/src/base/MooseApp.C

```diff
@@ -364,11 +364,16 @@ MooseApp::validParams()
   params.addParam<bool>(
       "automatic_automatic_scaling", false, "Whether to turn on automatic scaling by default");
 
-  MooseEnum libtorch_device_type("cpu cuda mps", "cpu");
+  MooseEnum compute_device_type("cpu cuda mps", "cpu");
```

Comment:
```suggestion
  MooseEnum compute_device_type("cpu cuda mps hip ceed-cpu ceed-cuda ceed-hip", "cpu");
```

Reply:
we'll need these for MFEM. See #31126 and https://github.com/idaholab/moose/pull/31151